### PR TITLE
feat: add support for js-ipfs dag api and also some tests

### PR DESCRIPTION
**Pull request by achingbrain**
Also support for loading raw blocks.

### CODE REVIEW

File: test/dag.spec.js [hunk 1]
```diff
@@ -70,16 +70,75 @@ describe('.dag', function () {
...
+    }, (err, block) => {
+      expect(err).to.not.exist()
+
+      ipfs.dag.get(block.cid, (err, result) => {
         expect(result).to.not.exist()
-        expect(err.message).to.equal('ipfs-http-client is missing DAG resolver for "raw" multicodec')
+        expect(err.message).to.equal('Missing IPLD format "git-raw"')
         done()
       })
     })
```
Comment (alanshaw) [inline test/dag.spec.js]: Can these be moved (and deduped) into `interface-ipfs-core`?

Comment (achingbrain) [reply test/dag.spec.js]: Sure.  This comes up a lot - what's the criteria on deciding where to put them?

Comment (alanshaw) [reply test/dag.spec.js]: Sorry, I missed this - I mostly view it as a place where we put core tests that are shared between `js-ipfs` and `js-ipfs-http-client`.

**Comment by alanshaw** [comment]
I'm waiting for these tests to pop up again in `interface-ipfs-core`...or did I miss a PR?

**Comment by alanshaw** [comment]
@achingbrain I've rebased and added the tests to `interface-ipfs-core`

**Pull request merged**